### PR TITLE
feat: add pypi-info command to retrieve package information from PyPI

### PR DESCRIPTION
**Pull request by EmilienM**
Add a new command 'pypi-info' that queries the PyPI JSON API to retrieve
comprehensive package information including:
- Package name, version, and availability status
- License information
- Homepage and repository URLs
- Distribution type availability (sdist/wheel)

The command supports both latest version queries and specific version
lookups with proper error handling for missing packages/versions.

Example usage:
  fromager pypi-info requests
  fromager pypi-info "requests==2.32.0"
  fromager pypi-info --pypi-base-url https://custom.pypi.org/pypi requests

Example of output:
```
$ fromager pypi-info requests
23:04:52 INFO Fetching information for requests
Package: requests
Version: 2.32.5
Found on PyPI: Yes
License: Apache-2.0
Homepage: https://requests.readthedocs.io
Repository: https://github.com/psf/requests
Has source distribution (sdist): Yes
Has wheel: Yes
23:04:52 INFO Package information retrieved successfully for requests 2.32.5
```

Includes comprehensive unit tests (13 test cases) and e2e integration tests
covering all functionality including error conditions and edge cases.

Co-Authored-By: Claude <noreply@anthropic.com> claude-sonnet-4@20250514
Signed-off-by: Emilien Macchi <emacchi@redhat.com>

### CODE REVIEW

**Review by tiran — CHANGES_REQUESTED** [review]
I had a similar idea. Let's get together on Monday.

File: src/fromager/commands/pypi_info.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        for url_type, url in project_urls.items():
+            if url_type.lower() in ("repository", "source", "github", "gitlab"):
+                print(f"Repository: {url}")
+                break
+
+    # Distribution types analysis
+    has_sdist = False
+    has_wheel = False
+
+    for url_info in urls:
```
Comment (dhellmann) [inline src/fromager/commands/pypi_info.py]: On recurring problem we have is packages with only prerelease versions, which are not resolved by default. It would be useful to show that sort of detail here when no version is specified.

**Comment by EmilienM** [comment]
I'm closing the PR, this was experimental. This work can be restored later.
